lib/basket.rb

```diff
@@ -0,0 +1,56 @@
+require 'time'
+require 'message'
+
+class Basket
+  attr_reader :contents, :total_price
+  ERROR_MESSAGE = "Please ensure the payment meets the order total"
+
+  def initialize(message = Message.new)
+    @contents = []
+    @message = message
+  end
+
+  def add(item, quantity)
+    quantity.times { @contents << item }
+  end
+
+  #Follow up with coach: how to best split out the reformatting of the hash from this method/
+  #is it ok for it to just live there
+  def view
+    h = Hash.new 0
+    @contents.each {|p| h[p] += 1}
+    @contents = h.keys.map{|k| k[:count] = h[k]; k}
+    @contents.each do |item|
+      price = item[:price] * item[:count]
+      puts "#{item[:pizza]} x#{item[:count]}, £#{price}"
+    end
+  end
+
+  def total 
```

Comment:
It's a small improvement but I feel as though you can probably use inject here to avoid needing the accumulator variable and the iteration.